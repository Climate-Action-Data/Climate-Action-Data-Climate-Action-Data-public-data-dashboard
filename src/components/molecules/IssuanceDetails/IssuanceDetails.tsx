import { Link, SimpleGrid, Skeleton } from '@chakra-ui/react'
import { useTranslation } from 'react-i18next'

import { DetailWidget } from '@/components/atoms/DetailWidget/DetailWidget'
import { formatDate } from '@/utils/DateFormat'
import { DateFormats } from '@/@types/DateFormats'
import { Issuance } from '@/@types/Issuance'
import { extractTagItemsFromTag } from '@/utils/TextConverter'
import { ExpandableList } from '@/components/atoms/ExpandableList/ExpandableList'
import { generateProjectUrl } from '@/utils/RequestHelpers'
import { SearchFlow } from '@/@types/Search'

interface IssuanceDetailsProps {
  issuance?: Issuance
}

const DEFAULT_COLUMN_MOBILE = 2
const DEFAULT_COLUMN_WEB = 4

export const IssuanceDetails = (props: IssuanceDetailsProps) => {
  const { issuance } = props
  const { t } = useTranslation(`projectDetails`)
  const { t: tHome } = useTranslation(`home`)
  if (!issuance) {
    return <Skeleton height="100px" />
  }

  const renderMonitoringDate = (monitoringPeriodStart: string, monitoringPeriodEnd: string) => {
    if (monitoringPeriodStart && monitoringPeriodEnd) {
      return `${formatDate(monitoringPeriodStart, DateFormats.YYYY_MM_DD)} - ${formatDate(monitoringPeriodEnd, DateFormats.YYYY_MM_DD)}`
    } else {
      return tHome(`noData`)
    }
  }

  return (
    <SimpleGrid columns={[DEFAULT_COLUMN_MOBILE, DEFAULT_COLUMN_WEB]} gap="24px">
      <DetailWidget title={t(`issuanceDetails.issuedTo`)}>
        <Link href={`${generateProjectUrl(issuance.project.warehouseProjectId, SearchFlow.UNIT)}`} textDecoration="underline">
          {issuance.project.name}
        </Link>
      </DetailWidget>
      <DetailWidget title={t(`issuanceDetails.vintage`)}>{issuance.vintage}</DetailWidget>
      <DetailWidget title={t(`issuanceDetails.quantity`)}>{issuance.quantity.toLocaleString()}</DetailWidget>
      <DetailWidget title={t(`issuanceDetails.unitType`)}>{issuance.unitType}</DetailWidget>
      <DetailWidget title={t(`issuanceDetails.monitoringPeriod`)}>{renderMonitoringDate(issuance?.monitoringPeriodStart, issuance?.monitoringPeriodEnd)}</DetailWidget>
      <DetailWidget title={t(`issuanceDetails.correspodinglyAdjusted`)}>{issuance.correspondingAdjustment ?? tHome(`noData`)}</DetailWidget>
      <DetailWidget asBox title={t(`issuanceDetails.issuanceTags`)}>
        {issuance.issuanceTags && issuance.issuanceTags.length > 0 ? <ExpandableList items={extractTagItemsFromTag(issuance.issuanceTags)} /> : tHome(`noData`)}
      </DetailWidget>
    </SimpleGrid>
  )
}
