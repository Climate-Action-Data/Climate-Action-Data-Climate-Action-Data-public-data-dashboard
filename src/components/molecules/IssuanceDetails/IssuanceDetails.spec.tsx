import { render } from '@testing-library/react'
import { IssuanceDetails } from './IssuanceDetails'
import { issuanceDetail } from '@/test/mock-data/issuance_detail'

it(`renders correctly with no project`, () => {
  const { container } = render(<IssuanceDetails issuance={undefined} />)
  expect(container).toMatchSnapshot()
})

it(`renders correctly with validation`, () => {
  const { container } = render(<IssuanceDetails issuance={issuanceDetail} />)
  expect(container).toMatchSnapshot()
})
