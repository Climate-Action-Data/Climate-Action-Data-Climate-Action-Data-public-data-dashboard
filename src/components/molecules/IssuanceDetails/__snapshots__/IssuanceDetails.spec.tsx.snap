// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no project 1`] = `
<div>
  <div
    class="chakra-skeleton css-1p3r0co"
  />
</div>
`;

exports[`renders correctly with validation 1`] = `
<div>
  <div
    class="css-cwfjkj"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Project Issued To
      </p>
      <p
        class="chakra-text css-0"
      >
        South Pole Carbon Asset Management
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Vintage
      </p>
      <p
        class="chakra-text css-0"
      >
        2016
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Quantity
      </p>
      <p
        class="chakra-text css-0"
      >
        195
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Unit Type
      </p>
      <p
        class="chakra-text css-0"
      >
        Reduction - nature
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Monitoring Period
      </p>
      <p
        class="chakra-text css-0"
      >
        2010/04/30 - 2020/03/04
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Correspondingly Adjusted
      </p>
      <p
        class="chakra-text css-0"
      >
        --
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Issuance Tags
      </p>
      <p
        class="chakra-text css-0"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            More information in the project link.
          </p>
          <p
            class="chakra-text css-0"
          >
            Cercarbono Standard.
          </p>
          <p
            class="chakra-text css-0"
          >
            Ecoregistry Serial Number of Issuance: CDC_12_2_1_322_14_CO_1_1_2019 .
          </p>
          <p
            class="chakra-text css-0"
          >
            Sector of issuance: Agriculture; forestry and fishing .
          </p>
          <button
            class="chakra-button css-4xx2wk"
            data-testid="verification-expand"
            type="button"
          >
            View More
          </button>
        </div>
      </p>
    </div>
  </div>
</div>
`;
